test/type/opaque.swift

```diff
@@ -590,3 +590,24 @@ func f62787_1(x: Bool) -> Optional<some Collection<Int>> {
   } 
   return nil // expected-error{{underlying type for opaque result type 'Optional<some Collection<Int>>' could not be inferred from return expression}}
 }
+
+// rdar://124482122 - Make sure that constraints are respected by opaque types
+protocol P3<A> {
+  associatedtype A: P1
+}
+
+do {
+  struct G<A: P1>: P3 {}
+
+  struct S: P1 {}
+
+  func test1() -> some P3<Int> { // expected-note {{opaque return type declared here}}
+    return G<S>()
+    // expected-error@-1 {{type of local function 'test1()' requires that 'S' conform to 'Int'}}
```

Comment:
do you want to fix the diagnostic to take requirement kind into account too? :)  It seems it was hardcoded to `RequirementKind::Conformance`

Reply:
Yeah, I'll look into that!